File: Low Level Design (LLD)/DesignPatterns/2. Behavariol/Bridge/Java/RemoteControl.java
public class RemoteControl {
    protected Device device;

    public RemoteControl(Device device) {
        this.device = device;
    }

    public void turnDeviceOn() {
        device.turnOn();
    }

    public void turnDeviceOff() {
        device.turnOff();
    }

    public void setDeviceVolume(int percent) {
        device.setVolume(percent);
    }
}

File: Low Level Design (LLD)/DesignPatterns/2. Behavariol/Bridge/Cpp/src/RemoteControl.cpp
#include "../include/RemoteControl.h"

RemoteControl::RemoteControl(Device* dev) : device(dev) {}

void RemoteControl::turnDeviceOn() {
    device->turnOn();
}

void RemoteControl::turnDeviceOff() {
    device->turnOff();
}

void RemoteControl::setDeviceVolume(int percent) {
    device->setVolume(percent);
}
